### DISCUSSION TITLE
How to migrate existing extensions to Montoya?

### DISCUSSION THREAD

**Discussion by chopicalqui** [PortSwigger/burp-extensions-montoya-api#16] [Q&A] [answered]
Dear PortSwigger Team

I want to migrate my existing extension [Turbo Data Miner](https://portswigger.net/bappstore/84350b8f090a4388a9d12cca141f201b) to Montoya but I would like to know the recommended way of doing it. In my opinion, we have two options:
1. We submit a new app.
   * Advantage: The current version could still be used by Burp Suite versions before 2022.11.
   * Disadvantage: There will be two versions in the app store, which will confuse users.
3. We just upgrade my existing app.
   * Disadvantage: Turbo Data Miner breaks for all Burp Suite versions before 2022.11.

Does PortSwigger has a recommendation/preference of how to proceed?

Thanks!

**Comment by Hannah-PortSwigger** [comment, marked as answer]
Thank you for your post.

When users attempt to install a BApp that is incompatible with their version of Burp, they will be prompted to upgrade their Burp version. This is part of the mechanism in the BappManifest file.

We also would like to avoid having multiple versions of a single extension in the BApp Store. Therefore, out of these two options, we would recommend updating your existing extension.

It is possible to write an extension using both the Extender API and the new Montoya API. You can find an example [here](https://github.com/PortSwigger/burp-extensions-montoya-api/tree/main/examples/src/main/java/net/portswigger/burp/extensions/multiapi). Rather than entirely rewriting your extension in the new Montoya API, it would be possible to make an extension that is compatible with older versions of Burp while still using the new Montoya API. This would add additional functionality in later versions of Burp.

Please bear in mind that the MontoyaApi parameter will not be passed in when the extension is loaded on older Burp versions, so you will need to include a null check to handle this case. We will be writing an example extension to demonstrate this in the future.
